src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -46,6 +46,44 @@ public void Clear(IEnumerable<int> source)
             Assert.True(s_empty == source.ToImmutableArray().Clear());
         }
 
+        [Theory]
+        [MemberData(nameof(Int32EnumerableData))]
+        public void AsSpanRoundTripTests(IEnumerable<int> source)
```

Comment:
nit: Should we add tests for calling `AsSpan()` on  `default` instances of `ImmutableArray<int>`?

Similarly, how about for T = reference type like string (i.e. `ImmutableArray<string>`)?